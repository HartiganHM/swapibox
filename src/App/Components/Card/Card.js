import React from 'react';
import './Card.css';

const Card = () => {
    return(
        <div className='Card'>
            Me a card!
        </div>
    )
}

export default Card;