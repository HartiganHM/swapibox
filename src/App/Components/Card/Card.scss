@import '../../../variables.scss';

.Card {
    background-color: $white;
    border: $border-card;
    border-radius: 3rem 0 3rem 0;
    color: $black;
    margin: 1rem;
    max-height: 30rem;

    .card-header {
        align-items: center;
        display: flex;
        border-bottom: 0.2rem groove $background-dark;
        font-size: 1.8rem;
        justify-content: space-around;

        .card-title {
            flex: 2;
        }

        .mark-favorite {
            background-color: $white;
            border: none;
            border-left: 0.2rem groove $background-dark;
            height: 4rem;
            width: 4rem;
        }
    }

    .card-info {
        display: flex;
        flex-direction: column;
        justify-content: space-between;

        .card-data-set {
            padding: $padding-small;
        }
    }
}