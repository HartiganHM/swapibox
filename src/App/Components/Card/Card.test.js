import React from 'react';
import Card from './Card';
import { shallow } from 'enzyme';

describe('Card Tests', () => {
  let renderedCard;
  let mockDisplayData;
  let mockCurrentFavorites;
  let mockSaveFavorites;
  let mockRemoveFavorites;

  beforeEach(() => {
    mockDisplayData = {
      name: 'Luke Skywalker',
      list: {
        Homeworld: 'Tatooine',
        Species: 'Human',
        Language: 'Intergalactic Basic',
        Population: '20000000'
      }
    };
    mockCurrentFavorites = [];
    mockSaveFavorites = jest.fn();
    mockRemoveFavorites = jest.fn();

    renderedCard = shallow(
      <Card
        displayData={mockDisplayData}
        currentFavorites={mockCurrentFavorites}
        saveFavorite={mockSaveFavorites}
        removeFavorite={mockRemoveFavorites}
      />
    );
  });

  it('Should exist', () => {
    expect(renderedCard).toBeDefined();
  });

  it('Should match the snapshot', () => {
    expect(renderedCard).toMatchSnapshot();
  });

  it('Should have have a list item for each data point', () => {
    expect(renderedCard.find('.card-data-set').length).toEqual(4);
  });

  it('Should have selected-favorite if card exists in favorites', () => {
    mockCurrentFavorites = [mockDisplayData];
    renderedCard = shallow(
      <Card
        displayData={mockDisplayData}
        currentFavorites={mockCurrentFavorites}
        saveFavorite={mockSaveFavorites}
      />
    );

    expect(renderedCard.find('.selected-favorite').length).toEqual(1);
  });

  it('Should have a removeFavorite button if in favorites', () => {
    expect(renderedCard.find('.current-favorite').length).toEqual(0);

    mockCurrentFavorites = [mockDisplayData];
    renderedCard = shallow(
      <Card
        displayData={mockDisplayData}
        currentFavorites={mockCurrentFavorites}
        saveFavorite={mockSaveFavorites}
      />
    );

    expect(renderedCard.find('.current-favorite').length).toEqual(1);
  });

  it('Should show a 0 for any list item with no value', () => {
    mockDisplayData = {
      name: 'Luke Skywalker',
      list: {
        Population: ''
      }
    };
    renderedCard = shallow(
      <Card
        displayData={mockDisplayData}
        currentFavorites={mockCurrentFavorites}
        saveFavorite={mockSaveFavorites}
      />
    );

    expect(renderedCard.find('.card-data-set').text()).toEqual('Population: 0');
  });
});
