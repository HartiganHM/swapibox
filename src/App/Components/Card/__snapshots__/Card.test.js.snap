// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Tests Should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    currentFavorites={Array []}
    displayData={
        Object {
            "list": Object {
              "Homeworld": "Tatooine",
              "Language": "Intergalactic Basic",
              "Population": "20000000",
              "Species": "Human",
            },
            "name": "Luke Skywalker",
          }
    }
    removeFavorite={[Function]}
    saveFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-header"
>
          <span
                    className="card-title"
          >
                    Luke Skywalker
          </span>
          <div
                    className="button-container"
          >
                    <button
                              className="mark-favorite"
                              onClick={[Function]}
                    />
          </div>
</div>,
        <ul
          className="card-info"
>
          <li
                    className="card-data-set"
          >
                    <span
                              className="data-key"
                    >
                              Homeworld
                    </span>
                    :
                     
                    Tatooine
          </li>
          <li
                    className="card-data-set"
          >
                    <span
                              className="data-key"
                    >
                              Species
                    </span>
                    :
                     
                    Human
          </li>
          <li
                    className="card-data-set"
          >
                    <span
                              className="data-key"
                    >
                              Language
                    </span>
                    :
                     
                    Intergalactic Basic
          </li>
          <li
                    className="card-data-set"
          >
                    <span
                              className="data-key"
                    >
                              Population
                    </span>
                    :
                     
                    20000000
          </li>
</ul>,
      ],
      "className": "Card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="card-title"
>
              Luke Skywalker
</span>,
            <div
              className="button-container"
>
              <button
                            className="mark-favorite"
                            onClick={[Function]}
              />
</div>,
          ],
          "className": "card-header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Luke Skywalker",
              "className": "card-title",
            },
            "ref": null,
            "rendered": "Luke Skywalker",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="mark-favorite"
                onClick={[Function]}
/>,
              "className": "button-container",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "mark-favorite",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "button",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="card-data-set"
>
              <span
                            className="data-key"
              >
                            Homeworld
              </span>
              :
               
              Tatooine
</li>,
            <li
              className="card-data-set"
>
              <span
                            className="data-key"
              >
                            Species
              </span>
              :
               
              Human
</li>,
            <li
              className="card-data-set"
>
              <span
                            className="data-key"
              >
                            Language
              </span>
              :
               
              Intergalactic Basic
</li>,
            <li
              className="card-data-set"
>
              <span
                            className="data-key"
              >
                            Population
              </span>
              :
               
              20000000
</li>,
          ],
          "className": "card-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "Homeworld",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="data-key"
>
                  Homeworld
</span>,
                ":",
                " ",
                "Tatooine",
              ],
              "className": "card-data-set",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Homeworld",
                  "className": "data-key",
                },
                "ref": null,
                "rendered": "Homeworld",
                "type": "span",
              },
              ":",
              " ",
              "Tatooine",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "Species",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="data-key"
>
                  Species
</span>,
                ":",
                " ",
                "Human",
              ],
              "className": "card-data-set",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Species",
                  "className": "data-key",
                },
                "ref": null,
                "rendered": "Species",
                "type": "span",
              },
              ":",
              " ",
              "Human",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "Language",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="data-key"
>
                  Language
</span>,
                ":",
                " ",
                "Intergalactic Basic",
              ],
              "className": "card-data-set",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Language",
                  "className": "data-key",
                },
                "ref": null,
                "rendered": "Language",
                "type": "span",
              },
              ":",
              " ",
              "Intergalactic Basic",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "Population",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span
                  className="data-key"
>
                  Population
</span>,
                ":",
                " ",
                "20000000",
              ],
              "className": "card-data-set",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Population",
                  "className": "data-key",
                },
                "ref": null,
                "rendered": "Population",
                "type": "span",
              },
              ":",
              " ",
              "20000000",
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-header"
>
            <span
                        className="card-title"
            >
                        Luke Skywalker
            </span>
            <div
                        className="button-container"
            >
                        <button
                                    className="mark-favorite"
                                    onClick={[Function]}
                        />
            </div>
</div>,
          <ul
            className="card-info"
>
            <li
                        className="card-data-set"
            >
                        <span
                                    className="data-key"
                        >
                                    Homeworld
                        </span>
                        :
                         
                        Tatooine
            </li>
            <li
                        className="card-data-set"
            >
                        <span
                                    className="data-key"
                        >
                                    Species
                        </span>
                        :
                         
                        Human
            </li>
            <li
                        className="card-data-set"
            >
                        <span
                                    className="data-key"
                        >
                                    Language
                        </span>
                        :
                         
                        Intergalactic Basic
            </li>
            <li
                        className="card-data-set"
            >
                        <span
                                    className="data-key"
                        >
                                    Population
                        </span>
                        :
                         
                        20000000
            </li>
</ul>,
        ],
        "className": "Card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="card-title"
>
                Luke Skywalker
</span>,
              <div
                className="button-container"
>
                <button
                                className="mark-favorite"
                                onClick={[Function]}
                />
</div>,
            ],
            "className": "card-header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Luke Skywalker",
                "className": "card-title",
              },
              "ref": null,
              "rendered": "Luke Skywalker",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="mark-favorite"
                  onClick={[Function]}
/>,
                "className": "button-container",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "mark-favorite",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="card-data-set"
>
                <span
                                className="data-key"
                >
                                Homeworld
                </span>
                :
                 
                Tatooine
</li>,
              <li
                className="card-data-set"
>
                <span
                                className="data-key"
                >
                                Species
                </span>
                :
                 
                Human
</li>,
              <li
                className="card-data-set"
>
                <span
                                className="data-key"
                >
                                Language
                </span>
                :
                 
                Intergalactic Basic
</li>,
              <li
                className="card-data-set"
>
                <span
                                className="data-key"
                >
                                Population
                </span>
                :
                 
                20000000
</li>,
            ],
            "className": "card-info",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "Homeworld",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="data-key"
>
                    Homeworld
</span>,
                  ":",
                  " ",
                  "Tatooine",
                ],
                "className": "card-data-set",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Homeworld",
                    "className": "data-key",
                  },
                  "ref": null,
                  "rendered": "Homeworld",
                  "type": "span",
                },
                ":",
                " ",
                "Tatooine",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "Species",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="data-key"
>
                    Species
</span>,
                  ":",
                  " ",
                  "Human",
                ],
                "className": "card-data-set",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Species",
                    "className": "data-key",
                  },
                  "ref": null,
                  "rendered": "Species",
                  "type": "span",
                },
                ":",
                " ",
                "Human",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "Language",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="data-key"
>
                    Language
</span>,
                  ":",
                  " ",
                  "Intergalactic Basic",
                ],
                "className": "card-data-set",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Language",
                    "className": "data-key",
                  },
                  "ref": null,
                  "rendered": "Language",
                  "type": "span",
                },
                ":",
                " ",
                "Intergalactic Basic",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": "Population",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span
                    className="data-key"
>
                    Population
</span>,
                  ":",
                  " ",
                  "20000000",
                ],
                "className": "card-data-set",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Population",
                    "className": "data-key",
                  },
                  "ref": null,
                  "rendered": "Population",
                  "type": "span",
                },
                ":",
                " ",
                "20000000",
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
